### PR TITLE
change acl permission strings to DRF permission constants

### PR DESCRIPTION
**Pull request by eviljeff**
start of #3572 
Only a sample of usage converted.

aims/ideas:
constants for permissions
make permission constants subclass `BasePermission` so can be used in DRF
easily allow them to be used as view decorators

### CODE REVIEW

**Comment by eviljeff** [comment]
Just throwing some ideas at the wall about what we can do with permissions, with the aim of (at least) having constants for acl `app:action` strings... if not unifying DRF Permissions, action_allowed usage and decorators for non-DRF views.

Not sure about the naming, or if `ACLPermission` should be in access or in api like the `GroupPermission` it replaces.

**Comment by EnTeQuAk** [comment]
just flying over the code, so very random comment, will give it a better review / more ideas etc a bit later. `permissions` should be named `perms` to make imports and code easier to read analogue to `acl` :)

**Comment by diox** [comment]
I like it. Not a huge fan of having the permissions in uppercase since they are more than just a constant but that's just nitpicking.

File: src/olympia/access/acl.py [hunk 1]
```diff
@@ -71,9 +72,9 @@ def check_collection_ownership(request, collection, require_owner=False):
     if not request.user.is_authenticated():
         return False
 
-    if action_allowed(request, 'Admin', '%'):
+    if permissions.ADMIN.has_permission(request):
         return True
-    elif action_allowed(request, 'Collections', 'Edit'):
+    elif permissions.COLLECTIONS_EDIT.has_permission(request):
```
Comment (EnTeQuAk) [inline src/olympia/access/acl.py]: Just thinking loud, I think I would like an API that looks like

```python
from olympia.access import perms # or `permissions as perms` as a general guideline

if perms.has_permission(request, perms.COLLECTIONS_EDIT):
    # ...
```

or generally how `action_allowed` looked like. It's just a feeling though, I can't explain what I dislike about it.

Comment (eviljeff) [inline src/olympia/access/acl.py]: making it the constants just tuples would still achieve the aims of consistently using permissions; decorator and DRF compatibility could be wrappers, used when needed.

**Comment by eviljeff** [comment]
gonna re-submit this (probably with constants as tuples) when I have more time.

**Pull request closed without merging**